Add IPv6 support to the Downward e2e API test

aojea opened the pull request:
Ref: https://github.com/kubernetes/kubernetes/issues/70248

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/release.md#issue-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Current regex used in the Downward e2e API tests is matching only IPv4 addresses, consequently, those tests fail with IPv6 clusters.

This patch modifies the regex to match ipv4 and ipv6 addresses.
**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes partially #70248 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note

```

Commit:
Add IPv6 support to the Downward e2e API test

Current regex used in the Downward e2e API tests is matching only
IPv4 addresses, consequently those tests fails with IPv6 clusters.

This patch modifies the regex to match ipv4 and ipv6 addresses.

Ref: https://github.com/kubernetes/kubernetes/issues/70248

aojea commented:
/sig node

aojea commented:
/assign @MrHohn

MrHohn commented:
/ok-to-test
Adding folks that might be better fit to review/approve this.
/assign @leblancd @rpothier

aojea commented:
@MrHohn seems those users can't be assigned, let me try with @oomichi
/assign @oomichi

oomichi commented on test/e2e/common/downward_api.go:
```diff
@@ -78,7 +78,7 @@ var _ = Describe("[sig-node] Downward API", func() {
 			fmt.Sprintf("POD_NAME=%v", podName),
 			fmt.Sprintf("POD_NAMESPACE=%v", f.Namespace.Name),
-			"POD_IP=(?:\\d+)\\.(?:\\d+)\\.(?:\\d+)\\.(?:\\d+)",
+			"POD_IP=(?:\\d+\\.\\d+\\.\\d+\\.\\d+|[a-fA-F0-9:]+)",
```
This pattern seems a little loose because it matches "TEST" also which is not IP address.
We might want to have more strict pattern as a test.
BTW, I guess we will re-use this pattern for IPv6 support in near future. If so, it is nice to have this pattern in `test/e2e/framework`.

aojea replied:
@oomichi I added a new regex and made it available as a constant in the framework pkg.
 I couldn´t find a simpler regex, that wasn´t too loose or that covered all the IPv6 addresses

aojea committed:
Add more strict IPv6 regex

Previous IPv6 regex was too loose, this patchs adds a better and
more strict regex for IPv6 addresses and makes the IPv4 and IPv6
regex availables as constants inside the framework pkg

aojea commented:
/assign @Random-Liu

oomichi commented:
/release-note-none

oomichi commented:
I confirmed several patterns of IPv6 (full form, drop leading zeroes, localhost, etc) can be validated with this code. Thanks for doing this.

/lgtm

MrHohn reviewed (approved):
/approve
